rocAL/rocAL/source/api/rocal_api_meta_data.cpp

```diff
@@ -77,12 +77,13 @@ ROCAL_API_CALL rocalCreateVideoLabelReader(RocalContext p_context, const char* s
 }
 
 RocalMetaData
-ROCAL_API_CALL rocalCreateCOCOReader(RocalContext p_context, const char* source_path, bool is_output){
+ROCAL_API_CALL rocalCreateCOCOReader(RocalContext p_context, const char* source_path, bool is_output, const bool mask) {
     if (!p_context)
         THROW("Invalid rocal context passed to rocalCreateCOCOReader")
     auto context = static_cast<Context*>(p_context);
-
-    return context->master_graph->create_coco_meta_data_reader(source_path, is_output, MetaDataReaderType::COCO_META_DATA_READER,  MetaDataType::BoundingBox);
+    if(mask)
+        return context->master_graph->create_coco_meta_data_reader(source_path, is_output, MetaDataReaderType::COCO_META_DATA_READER, MetaDataType::PolygonMask);
```

Comment:
can't we combine both mask and bounding bix reading in one single function. Going through the dataset twice is very inefficient

Reply:
It is also not ok to THROW exception in an API without a try catch loop. You can return a invalid return value if the context is not valid instead of throw()

Reply:
1. The coco_meta_data_reader reads both polygon masks and bboxes in a single read
2. The THROW exception was already present in this function and all the other reader creation functions